src/worker/lambda/handlerPuller.go

```diff
@@ -106,14 +105,41 @@ func (cp *HandlerPuller) isRemote() bool {
 	return strings.HasPrefix(cp.prefix, "http://") || strings.HasPrefix(cp.prefix, "https://")
 }
 
+func createRegex(expr string, testValidString string, testInvalidString string) (reComp *regexp.Regexp, reErr error) {
+	re, err := regexp.Compile(expr)
+	errorMsg := error(nil)
+
+	if err != nil {
+		msg := "regex failed to comple with error '%s'"
+		errorMsg = fmt.Errorf(msg, err)
+	}
+	if re == nil {
+		errorMsg = errors.New("regex failed to compile with null output")
+	} else {
+		vMatched := re.MatchString(testValidString)
+		if !vMatched {
+			errorMsg = errors.New("regex valid test run failed")
+		}
+		iMatched := re.MatchString(testInvalidString)
+		if iMatched {
+			errorMsg = errors.New("regex invalid test run failed")
+		}
+	}
+
+	return re, errorMsg
+}
+
 func (cp *HandlerPuller) Pull(name string) (rt_type common.RuntimeType, targetDir string, err error) {
 	t := common.T0("pull-lambda")
 	defer t.T1()
 
-	matched, err := regexp.MatchString(`^[A-Za-z0-9\.\-\_]+$`, name)
-	if err != nil {
-		return rt_type, "", err
-	} else if !matched {
+	handlerNameRegex, handlerNameError := createRegex(`^[A-Za-z0-9\.\-\_]+$`, "-H3ll0.world_", "Hello world!")
```

Comment:
this will compile a new regex every time, so we won't have any performance benefit.  We want to compile once and have it in a global variable or somewhere so that we can keep reusing with recompiling.